pkg/sysinfo/sysinfo_linux_test.go

```diff
@@ -56,3 +59,36 @@ func TestCgroupEnabled(t *testing.T) {
 		t.Fatal("cgroupEnabled should be true")
 	}
 }
+
+func TestNew(t *testing.T) {
+	sysInfo := New(false)
+	require.NotNil(t, sysInfo)
+
+	sysInfo = New(true)
+	require.NotNil(t, sysInfo)
+
+	// Check if AppArmor is supported. then it must be TRUE , else FALSE
+	if _, err := os.Stat("/sys/kernel/security/apparmor"); !os.IsNotExist(err) {
```

Comment:
It's slightly concerning that this test has different behavior depending whether AppArmor is enabled on the system or not. That may lead to the test giving different results depending on the system where it is run.

I think it might be better to split this into two tests, `TestNewAppArmorEnabled` and `TestNewAppArmorDisabled`. The former would call `t.Skip()` if AppArmor is disabled, and the latter would call `t.Skip()` if AppArmor is enabled. That way, it would be clear from the test output which path of the two was tested.